### PR TITLE
Implement Individual pages for Routing

### PR DESCRIPTION
**Pull request by Alee-star**
Implement Individual pages for Routing

### CODE REVIEW

File: src/components/Banner/GalleryList.jsx [hunk 1]
```diff
@@ -36,43 +39,35 @@ const ImageSection = () => {
...
-    )
-      ? currentFavourites.filter((fav) => fav.id !== photo.id)
-      : [...currentFavourites, photo];
-    saveFavInLocalStorage(updatedFavourites);
-    setPhotos((prevPhotos) =>
-      prevPhotos.map((p) =>
-        p.id === photo.id ? { ...p, favourite: !p.favourite } : p
-      )
+    const updatedFavourites = toggleFavourite(photo, currentFavourites);
+    const updatedPhotos = photos.map((p) =>
```
Comment (vonnue-nidheesh) [inline src/components/Banner/GalleryList.jsx]: use meaningful names. Use `photo` instead of `p`.

File: src/components/Banner/GalleryList.jsx [hunk 2]
```diff
@@ -36,43 +39,35 @@ const ImageSection = () => {
...
+      p.id === photo.id ? { ...p, favourite: !p.favourite } : p
     );
+    setPhotos(updatedPhotos);
   };
 
   const handleTabChange = (tab) => {
```
Comment (vonnue-nidheesh) [inline src/components/Banner/GalleryList.jsx]: Should we need this function and this props anymore?

File: src/components/Banner/GalleryList.jsx [hunk 3]
```diff
@@ -1,10 +1,10 @@
 import React, { useState, useEffect } from "react";
 import axios from "axios";
-import Navbar from "../Navbar/Navbar";
 import Favourites from "../Favourites/Favourites";
 import ImageCard from "./ImageCard";
 import "../Favourites/Favourites.css";
 import "./GalleryList.css";
+import { getFavFromLocalStorage, toggleFavourite } from "../../ToggleFunction";
```
Comment (vonnue-nidheesh) [inline src/components/Banner/GalleryList.jsx]: Sort imports

File: src/components/Banner/GalleryList.jsx [hunk 4]
```diff
@@ -36,43 +39,35 @@ const ImageSection = () => {
...
-      />
       <div className="image-section">
         {selectedNavItem === "Favourites" ? (
           <Favourites
-            photos={photos.filter((photo) => photo.favourite)}
-            addToFavourites={toggleFavourite}
+            photos={favouritePhotos}
+            addToFavourites={handleToggleFavourite}
           />
         ) : (
```
Comment (vonnue-nidheesh) [inline src/components/Banner/GalleryList.jsx]: Should we need this favourite section here?

File: src/components/Banner/GalleryList.jsx [hunk 5]
```diff
@@ -36,43 +39,35 @@ const ImageSection = () => {
...
           photos.map((photo) => (
             <ImageCard
               key={photo.id}
               photo={photo}
-              addToFavourites={toggleFavourite}
+              addToFavourites={handleToggleFavourite}
               IsLike={photo.favourite}
```
Comment (vonnue-nidheesh) [inline src/components/Banner/GalleryList.jsx]: `isLike`

File: src/pages/Home/Home.jsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import "./Home.css";
+
+const Home = () => {
+  return (
+    <div>
```
Comment (vonnue-nidheesh) [inline src/pages/Home/Home.jsx]: Why do we need an extra div here?

File: src/components/Banner/GalleryList.jsx [hunk 7]
```diff
@@ -28,6 +28,9 @@ const ImageSection = () => {
           favourite: getFavFromLocalStorage().some(
             (fav) => fav.id === photo.id
           ),
+          favourite: getFavFromLocalStorage().some(
+            (fav) => fav.id === photo.id
+          ),
```
Comment (vonnue-nidheesh) [inline src/components/Banner/GalleryList.jsx]: Why same prop in two times?

File: src/App.js [hunk 8]
```diff
@@ -1,27 +1,22 @@
...
 import { Route, Routes, Navigate } from "react-router-dom";
-import Videos from "./components/Videos";
-import Leaderboard from "./components/Leaderboard";
-import Challenges from "./components/Challenges";
 import Heading from "./components/Header/Heading";
-import FavouritePage from "./components/FavouritePage";
+import Home from "./pages/Home/Home";
+import Photos from "./pages/Photos";
+import FavouritePage from "./pages/FavouritePage";
+import Navbar from "./components/Navbar/Navbar";
```
Comment (vonnue-nidheesh) [inline src/App.js]: Sort imports

File: src/components/Banner/GalleryList.jsx [hunk 9]
```diff
@@ -28,6 +28,9 @@ const ImageSection = () => {
           favourite: getFavFromLocalStorage().some(
             (fav) => fav.id === photo.id
           ),
```
Comment (vonnue-nidheesh) [inline src/components/Banner/GalleryList.jsx]: Better to use includes array function here.

File: src/components/Banner/ImageCard.jsx [hunk 10]
```diff
@@ -5,7 +5,6 @@ import "./ImageCard.css";
 
 const ImageCard = ({ photo, IsLike, addToFavourites }) => {
   const [hovered, setHovered] = useState(false);
-
```
Comment (vonnue-nidheesh) [inline src/components/Banner/ImageCard.jsx:8]: Need this empty line here.

File: src/components/Banner/GalleryList.jsx (hunk 2 above)
Comment (Alee-star) [reply src/components/Banner/GalleryList.jsx]: sePhotos is necessary

Comment (Alee-star) [reply src/components/Banner/GalleryList.jsx]: handkeTabChange is removed

**Review by vonnue-nidheesh — APPROVED** [review]

**Pull request merged**